### PR TITLE
Provide split key in tablet reader

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JaySon-Huang — APPROVED** [review]
TRY

**Review by JaySon-Huang — DISMISSED** [review]
REVERT

File: proxy_components/mock-engine-store/Cargo.toml [hunk 1]
```diff
@@ -68,6 +68,7 @@ rand = "0.8"
 resolved_ts = { workspace = true }
 resource_control = { workspace = true }
 resource_metering = { workspace = true }
+serde_json = "1.0"
```
Comment (CalvinNeo) [inline proxy_components/mock-engine-store/Cargo.toml:71]: This is a test comment

**Review by Lloyd-Pottiger — COMMENTED** [review]
try

File: proxy_components/mock-engine-store/src/mock_cluster/cluster_ext.rs [hunk 2]
```diff
@@ -80,8 +80,11 @@ impl ClusterExt {
         cluster_ext_ptr: isize,
         mock_cfg: MockConfig,
         pd_client: Option<Arc<dyn pd_client::PdClient>>,
+        proxy_cfg: &ProxyConfig,
```
Comment (Lloyd-Pottiger) [inline proxy_components/mock-engine-store/src/mock_cluster/cluster_ext.rs:83]: try

File: proxy_components/mock-engine-store/src/mock_cluster/cluster_ext.rs [hunk 3]
```diff
@@ -96,6 +99,7 @@ impl ClusterExt {
             },
             None,
             pd_client,
+            proxy_config_str,
```
Comment (CalvinNeo) [inline proxy_components/mock-engine-store/src/mock_cluster/cluster_ext.rs:102]: try comment

**Pull request merged**